andela/ah-infinity-stones
# #161966929 Implement article read time feature

### What does this PR do?
Implement the article read time feature

### Description of Task to be completed
Estimate the amount of time it takes to read an article

### How should this be manually tested?
* `git clone -b ft-read-time-161966929 https://github.com/andela/ah-infinity-stones` and ` cd ah-infinity-stones`.
* Run `python3 -m venv virtual` to create a virtual environment and `source virtual/bin/activate` to activate it.
* `pip install -r requirements.txt` to install dependencies
* Create a test database in the postgres shell
* Set up environment variables using the sample below as a guide:
```export DATABASE_USER='<postgres username>'
export DATABASE_PASSWORD='<postgres user's password>'
export HOST='localhost'
export PORT='5432'
export DATABASE_NAME='<name_of_database>' 
export SECRET_KEY='<personal_secret_key>'
export EMAIL_HOST_USER='<email_host_username>'
export EMAIL_HOST_PASSWORD='<email_host_password>'
```
* Run `python3 manage.py makemigrations` followed by `python3 manage.py migrate` to apply migrations
 Acess `http://127.0.0.1:8000/api/users/` through [Postman](https://www.getpostman.com/), provide inputs in the following format and create account.
```
 {
	"user": {
		"username": "remmykn",
		"email": "remmy@test.com",
		"password": "Password@123"
	}
}

```
Check your email and activate the account. Use the registration token to access the following endpoints:
- `http://127.0.0.1:8000/api/articles` - to post an article 


### Relevant pivotal tracker stories
[Delivers #161966929](https://www.pivotaltracker.com/story/show/161966929)

### Screenshots
![image](https://user-images.githubusercontent.com/17976289/50598714-ab2cef00-0ebc-11e9-85e4-553fd92c3fb7.png)
![image](https://user-images.githubusercontent.com/17976289/50598802-ef1ff400-0ebc-11e9-9964-bf64c77d9057.png)
